### PR TITLE
Minor refactoring

### PR DESCRIPTION
**Pull request by pandeyharsh**
Trying to conform to the [Groovy Style Guide](http://www.groovy-lang.org/style-guide.html).

### CODE REVIEW

File: gradle/src/main/groovy/com/flurry/android/symbols/SymbolUploadPlugin.groovy [hunk 1]
```diff
@@ -22,11 +22,11 @@ class SymbolUploadPlugin implements Plugin<Project> {
     @Override
     void apply(Project project) {
         UploadProGuardMapping.logger = project.logger
-        getOrCreateConfig(project)
+        getOrCreateConfig project
```
Comment (clayreimann) [inline gradle/src/main/groovy/com/flurry/android/symbols/SymbolUploadPlugin.groovy]: Style guide be damned, parenthesis add needed clarity. I don't do it in Ruby or coffeescript and I disagree with removing parens here too.

File: gradle/src/main/groovy/com/flurry/android/symbols/SymbolUploadPlugin.groovy [hunk 2]
```diff
@@ -57,29 +57,30 @@ class SymbolUploadPlugin implements Plugin<Project> {
...
-        if (config == null) {
-            config = target.extensions.create(CONFIGURATION_KEY, SymbolUploadConfiguration.class)
+    private static SymbolUploadConfiguration getOrCreateConfig(Project target) {
+        SymbolUploadConfiguration config = target.extensions.findByType SymbolUploadConfiguration
+        if (!config) {
+            config = target.extensions.create CONFIGURATION_KEY, SymbolUploadConfiguration
         }
-        return config
+
+        config
```
Comment (clayreimann) [inline gradle/src/main/groovy/com/flurry/android/symbols/SymbolUploadPlugin.groovy]: I want my `return` statements too. `config` is a valid expression even when it's not the last line of the function. Without the explicit `return` the compiler won't catch it if someone adds code below this line, which will cause the return value of this function can change unexpectedly.

File: gradle/src/main/groovy/com/flurry/android/symbols/SymbolUploadPlugin.groovy [hunk 3]
```diff
@@ -57,29 +57,30 @@ class SymbolUploadPlugin implements Plugin<Project> {
...
+        Map<String, String> configValues = new HashMap<>()
 
         if (config.apiKey) {
             configValues[API_KEY] = config.apiKey
         }
         if (config.token) {
             configValues[TOKEN] = config.useEnvVar ? System.getenv(config.token) : config.token
         }
-        configValues.put(TIMEOUT, config.uploadTimeout)
+        configValues.put TIMEOUT, config.uploadTimeout as String
```
Comment (clayreimann) [inline gradle/src/main/groovy/com/flurry/android/symbols/SymbolUploadPlugin.groovy]: Put parens back so we don't need the explicit cast.

File: gradle/src/main/groovy/com/flurry/android/symbols/SymbolUploadPlugin.groovy [hunk 4]
```diff
@@ -57,29 +57,30 @@ class SymbolUploadPlugin implements Plugin<Project> {
...
 
-        if (config.configPath != null) {
-            configValues.putAll(UploadProGuardMapping.parseConfigFile(config.configPath))
+        if (config.configPath) {
+            configValues.putAll UploadProGuardMapping.parseConfigFile(config.configPath) as Map<? extends String, ? extends String>
```
Comment (clayreimann) [inline gradle/src/main/groovy/com/flurry/android/symbols/SymbolUploadPlugin.groovy]: Bad, bad cast

File: gradle/src/main/groovy/com/flurry/android/symbols/SymbolUploadPlugin.groovy [hunk 5]
```diff
@@ -57,29 +57,30 @@ class SymbolUploadPlugin implements Plugin<Project> {
...
         }
 
-        return configValues
+        configValues
```
Comment (clayreimann) [inline gradle/src/main/groovy/com/flurry/android/symbols/SymbolUploadPlugin.groovy]: Explicit return.

File: gradle/src/main/groovy/com/flurry/android/symbols/SymbolUploadPlugin.groovy (hunk 3 above)
Comment (pandeyharsh) [reply gradle/src/main/groovy/com/flurry/android/symbols/SymbolUploadPlugin.groovy]: Even with the parentheses, IntelliJ gives a warning. The explicit cast only gets rid of the warning and has nothing to do with the parens. I can put the parens back but the cast serves a purpose.

File: gradle/src/main/groovy/com/flurry/android/symbols/SymbolUploadPlugin.groovy (hunk 4 above)
Comment (pandeyharsh) [reply gradle/src/main/groovy/com/flurry/android/symbols/SymbolUploadPlugin.groovy]: Similarly, the explicit cast only gets rid of the warning.

**Pull request merged**